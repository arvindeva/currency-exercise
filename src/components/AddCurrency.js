import React from 'react';

const AddCurrency = props => {
  return (
    <div className="ui segment">
      <h1>Add Currency!</h1>
    </div>
  )
}

export default AddCurrency;