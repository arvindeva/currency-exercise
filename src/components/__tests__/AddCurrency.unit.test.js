import React from 'react';
import { shallow, mount } from 'enzyme';
import AddCurrency from '../AddCurrency';

describe('AddCurrency tests', () => {
  test('renders AddCurrency', () => {
    const wrapper = shallow(<AddCurrency />);
    expect(wrapper).toMatchSnapshot();
  });

  test('calls onClick event on click of the button', () => {
    const onAddClick = jest.fn();
    let wrapper = mount(<AddCurrency onAddClick={onAddClick} />);
    wrapper
      .find('button')
      .first()
      .simulate('click');
    expect(onAddClick).toHaveBeenCalled();
  });
});
