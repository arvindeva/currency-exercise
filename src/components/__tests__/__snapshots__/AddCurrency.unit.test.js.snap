// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCurrency tests renders AddCurrency 1`] = `
<styled.div
  className="ui segments"
>
  <div
    className="ui segment"
  >
    <h1>
      Add More Currencies!
    </h1>
  </div>
  <div
    className="ui segment content"
  >
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      className="dropdown"
      closeOnBlur={true}
      deburr={false}
      fluid={true}
      icon="dropdown"
      minCharacters={1}
      noResultsMessage="No results found."
      onChange={[Function]}
      openOnFocus={true}
      options={
        Array [
          Object {
            "key": "USD",
            "text": "USD - United States Dollars",
            "value": "USD",
          },
          Object {
            "key": "EUR",
            "text": "EUR - European Euro",
            "value": "EUR",
          },
          Object {
            "key": "CAD",
            "text": "CAD - Canadian Dollars",
            "value": "CAD",
          },
          Object {
            "key": "IDR",
            "text": "IDR - Indonesian Rupiah",
            "value": "IDR",
          },
          Object {
            "key": "GBP",
            "text": "GBP - Great Britain Pounds",
            "value": "GBP",
          },
          Object {
            "key": "CHF",
            "text": "CHF - Swiss Franc",
            "value": "CHF",
          },
          Object {
            "key": "SGD",
            "text": "SGD - Singapore Dollars",
            "value": "SGD",
          },
          Object {
            "key": "INR",
            "text": "INR - Indian Rupee",
            "value": "INR",
          },
          Object {
            "key": "MYR",
            "text": "MYR - Malaysian Ringgit",
            "value": "MYR",
          },
          Object {
            "key": "JPY",
            "text": "JPY - Japanese Yen",
            "value": "JPY",
          },
          Object {
            "key": "KRW",
            "text": "KRW - South Korean Won",
            "value": "KRW",
          },
        ]
      }
      placeholder="Select Currency"
      renderLabel={[Function]}
      searchInput="text"
      selectOnBlur={true}
      selectOnNavigation={true}
      selection={true}
      value=""
      wrapSelection={true}
    />
    <button
      className="ui button basic black"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</styled.div>
`;
